src/lib/image_hash/ahash_result.h

```diff
@@ -67,38 +70,38 @@ class ahash_result
 
 	std::array<uint64_t, 9> extract_all() const
 	{
-		bit_extractor<intx::uint128, 100> be(_bits);
+		bit_extractor<intx::uint128, CELLAREA, CELLSIZE> be(_bits);
 		return {
 			// top row -- top left bit is the start bit (0). bottom right is end bit.
-			be.extract(0, 10, 20, 30, 40, 50, 60, 70), // left
-			be.extract(1, 11, 21, 31, 41, 51, 61, 71),
-			be.extract(2, 12, 22, 32, 42, 52, 62, 72), // right
+			be.extract_tuple( be.pattern(0) ), // left
+			be.extract_tuple( be.pattern(1) ),
+			be.extract_tuple( be.pattern(2) ), // right
 			// middle row
-			be.extract(10, 20, 30, 40, 50, 60, 70, 80),
-			be.extract(11, 21, 31, 41, 51, 61, 71, 81),
-			be.extract(12, 22, 32, 42, 52, 62, 72, 82),
+			be.extract_tuple( be.pattern(3) ),
+			be.extract_tuple( be.pattern(4) ),
+			be.extract_tuple( be.pattern(5) ),
 			// bottom row
-			be.extract(20, 30, 40, 50, 60, 70, 80, 90),
-			be.extract(21, 31, 41, 51, 61, 71, 81, 91),
-			be.extract(22, 32, 42, 52, 62, 72, 82, 92)
+			be.extract_tuple( be.pattern(6) ),
+			be.extract_tuple( be.pattern(7) ),
+			be.extract_tuple( be.pattern(8) )
```

Comment:
These were already pretty magical -- to deal with configurable cell dimensions, they are now sadly even more magical.